$OpenBSD$

remove ucontext header (deprecated; not present)
stub a function relying on ucontext

Index: Utilities/Thread.cpp
--- Utilities/Thread.cpp.orig
+++ Utilities/Thread.cpp
@@ -27,7 +27,7 @@
 #endif
 #include <errno.h>
 #include <signal.h>
-#include <ucontext.h>
+typedef ucontext_t SContext;
 #include <pthread.h>
 #include <sys/time.h>
 #include <sys/resource.h>
@@ -797,30 +797,6 @@ register_t* freebsd_x64reg(x64_context *context, int r
 
 long* openbsd_x64reg(x64_context *context, int reg)
 {
-	auto *state = &context->uc_mcontext;
-	switch(reg)
-	{
-	case 0: return &state->sc_rax;
-	case 1: return &state->sc_rcx;
-	case 2: return &state->sc_rdx;
-	case 3: return &state->sc_rbx;
-	case 4: return &state->sc_rsp;
-	case 5: return &state->sc_rbp;
-	case 6: return &state->sc_rsi;
-	case 7: return &state->sc_rdi;
-	case 8: return &state->sc_r8;
-	case 9: return &state->sc_r9;
-	case 10: return &state->sc_r10;
-	case 11: return &state->sc_r11;
-	case 12: return &state->sc_r12;
-	case 13: return &state->sc_r13;
-	case 14: return &state->sc_r14;
-	case 15: return &state->sc_r15;
-	case 16: return &state->sc_rip;
-	default:
-		LOG_ERROR(GENERAL, "Invalid register index: %d", reg);
-		return nullptr;
-	}
 }
 
 #elif defined(__NetBSD__)
